### PR TITLE
fix(telephony.rma): add change type for RMA Unknown

### PR DESCRIPTION
**Pull request by ghyenne**
| Question         | Answer
| ---------------- | ---
| Branch?          | `develop` 
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | MANAGER-8121 
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~

## Description
I added change type for RMA Unknown

## Related
:flags:  Translations:

- [x] TRANS-44603

### CODE REVIEW

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/assist/rma/rma.constants.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+export const MAC_ADDRESS_FORMAT_REGEX = /\w{1,2}/g;
+
+export const RMA_URL_PREFIX =
+  'http://www.ovh.com/cgi-bin/telephony/rma.pl?reference=';
```
Comment (anooparveti) [inline packages/manager/apps/telecom/src/app/telecom/telephony/line/assist/rma/rma.constants.js:4]: **question:** Could you please check if you can use https. Also fixes the Security hotspot reported during SonarCloud analysis.
```suggestion
  'https://www.ovh.com/cgi-bin/telephony/rma.pl?reference=';
```

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/assist/rma/rma.controller.js [hunk 2]
```diff
@@ -1,101 +1,159 @@
...
-        serviceName: $stateParams.serviceName,
-      }).$promise;
-  };
+  modalDismiss() {
+    this.isOpenModal = false;
+    this.modalPrimaryAction = null;
+    this.modalTitle = null;
+    this.modalContent = null;
+    this.modalRma = null;
+  }
```
Comment (anooparveti) [inline packages/manager/apps/telecom/src/app/telecom/telephony/line/assist/rma/rma.controller.js:36]: **suggestion:** Can be renamed to `resetModalInfo` and use it on `$onInit` as well.

**Review by anooparveti — APPROVED** [review]

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/assist/rma/rma.controller.js [hunk 3]
```diff
@@ -1,101 +1,159 @@
...
-      .$promise.then(() => {
-        remove(self.rmaList, { id: rma.id });
-        TucToast.success(
-          $translate.instant('telephony_line_assist_rma_cancel_success'),
+  generateCancelRmaModal(rma) {
+    this.rmaTrackClick('rma-cancel');
+    this.modalRma = rma;
+    this.modalTitle = TEXT_FOR_MODAL.cancelRma;
+    this.modalContent = `${TEXT_FOR_MODAL.cancelRma}_modal_content`;
+    this.modalPrimaryAction = ACTION_TYPE.cancelRma;
```
Comment (mohammed-zahaf) [inline packages/manager/apps/telecom/src/app/telecom/telephony/line/assist/rma/rma.controller.js:47]: may be we can consider to use a modal object to group properties ?
```suggestion
    this.modal = {
    rma,
    title: TEXT_FOR_MODAL.cancelRma,
    content: `${TEXT_FOR_MODAL.cancelRma}_modal_content`,
    primaryAction: ACTION_TYPE.cancelRma,
    }
```

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/assist/rma/rma.controller.js [hunk 4]
```diff
@@ -1,101 +1,159 @@
...
+    this.modalPrimaryAction = ACTION_TYPE.keepingLine;
+    this.isOpenModal = true;
+  }
+
+  generateTerminateLineModal(rma) {
+    this.rmaTrackClick('rma-resiliate-line');
+    this.modalRma = rma;
+    this.modalTitle = TEXT_FOR_MODAL.terminateLine;
+    this.modalContent = `${TEXT_FOR_MODAL.terminateLine}_modal_content`;
+    this.modalPrimaryAction = ACTION_TYPE.terminateLine;
```
Comment (mohammed-zahaf) [inline packages/manager/apps/telecom/src/app/telecom/telephony/line/assist/rma/rma.controller.js:74]: same as here :
https://github.com/ovh/manager/pull/7325/files#r893440387

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/assist/rma/rma.html [hunk 5]
```diff
@@ -117,50 +119,102 @@ <h1 data-translate="telephony_line_assist_rma_title"></h1>
...
+                    data-translate="telephony_line_assist_rma_keeping_line"
+                ></button>
+                <!-- REPORTING EQUIPEMENT OUT OF ORDER -->
+                <button
+                    type="button"
+                    class="btn btn-default"
+                    data-ng-click="$ctrl.generateEquipmentOutOfOrderModal(rma)"
+                    data-translate="telephony_line_assist_rma_equipment_out_of_order"
+                ></button>
+                <!-- TEMINATE THE LINE -->
```
Comment (mohammed-zahaf) [inline packages/manager/apps/telecom/src/app/telecom/telephony/line/assist/rma/rma.html:182]: may be you would say `TERMINATE`.
```suggestion
                <!-- TERMINATE THE LINE -->
```

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/assist/rma/rma.html [hunk 6]
```diff
@@ -117,50 +119,102 @@ <h1 data-translate="telephony_line_assist_rma_title"></h1>
...
+                    class="btn btn-danger"
+                    data-translate="telephony_line_assist_rma_terminate_line"
+                    data-ng-click="$ctrl.generateTerminateLineModal(rma)"
+                ></button>
+            </dl>
+        </div>
+    </div>
+
+    <div data-ng-if="$ctrl.isOpenModal">
+        <div class="modal d-block" tabindex="-1">
```
Comment (mohammed-zahaf) [inline packages/manager/apps/telecom/src/app/telecom/telephony/line/assist/rma/rma.html:194]: Why a `tabindex="-1"` ?

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/assist/rma/translations/Messages_fr_FR.json [hunk 7]
```diff
@@ -30,7 +30,19 @@
...
-}
\ No newline at end of file
+  "telephony_line_assist_rma_pdf_download": "Télécharger le bon de retour",
+  "telephony_line_assist_rma_equipment_out_of_order": "Signaler un equipement hors service",
+  "telephony_line_assist_rma_keeping_line": "Conserver la ligne téléphonique sans matériel",
+  "telephony_line_assist_rma_terminate_line": "Résilier la ligne téléphonique associée",
+  "telephony_line_assist_rma_modal_yes": "Oui",
+  "telephony_line_assist_rma_modal_no": "Non",
+  "telephony_line_assist_rma_cancel_modal_content": "Voulez-vous annuler ce RMA ?",
+  "telephony_line_assist_rma_equipment_out_of_order_modal_content": "Voulez-vous signaler un equipement hors service ?{{ br }}(Vous serez alors redirigé vers la page de déclaration d'incident)",
```
Comment (mohammed-zahaf) [inline packages/manager/apps/telecom/src/app/telecom/telephony/line/assist/rma/translations/Messages_fr_FR.json:43]:
```suggestion
  "telephony_line_assist_rma_equipment_out_of_order_modal_content": "Voulez-vous signaler un équipement hors service ?{{ br }}(Vous serez alors redirigé vers la page de déclaration d'incident)",
```

Comment (mohammed-zahaf) [reply packages/manager/apps/telecom/src/app/telecom/telephony/line/assist/rma/translations/Messages_fr_FR.json:43]: `breakLine` could be more understandable.
```suggestion
  "telephony_line_assist_rma_equipment_out_of_order_modal_content": "Voulez-vous signaler un equipement hors service ?{{ breakLine }}(Vous serez alors redirigé vers la page de déclaration d'incident)",
```

Comment (ghyenne) [reply packages/manager/apps/telecom/src/app/telecom/telephony/line/assist/rma/translations/Messages_fr_FR.json]: I fixed.

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/assist/rma/translations/Messages_fr_FR.json [hunk 8]
```diff
@@ -30,7 +30,19 @@
...
   "telephony_line_assist_rma_step_name_parcelValidation": "Validation du colis",
   "telephony_line_assist_rma_step_name_validation": "Validation du changement de téléphone",
+  "telephony_line_assist_rma_step_name_dispatchJustification": "Justification de l'envoi",
   "telephony_line_assist_rma_step_status_todo": "En attente",
   "telephony_line_assist_rma_step_status_done": "Validé",
-  "telephony_line_assist_rma_pdf_download": "Télécharger le bon de retour"
-}
\ No newline at end of file
+  "telephony_line_assist_rma_pdf_download": "Télécharger le bon de retour",
+  "telephony_line_assist_rma_equipment_out_of_order": "Signaler un equipement hors service",
```
[the remark this answers is not in the document]
Comment (ghyenne) [reply packages/manager/apps/telecom/src/app/telecom/telephony/line/assist/rma/translations/Messages_fr_FR.json]: I fixed.

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/assist/rma/rma.html (hunk 6 above)
Comment (ghyenne) [reply packages/manager/apps/telecom/src/app/telecom/telephony/line/assist/rma/rma.html]: It's not necessary. I fixed.

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/assist/rma/rma.html (hunk 5 above)
Comment (ghyenne) [reply packages/manager/apps/telecom/src/app/telecom/telephony/line/assist/rma/rma.html]: I fixed.

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/assist/rma/rma.controller.js (hunk 4 above)
Comment (ghyenne) [reply packages/manager/apps/telecom/src/app/telecom/telephony/line/assist/rma/rma.controller.js]: I fixed.

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/assist/rma/rma.controller.js [hunk 9]
```diff
@@ -1,101 +1,159 @@
...
+    this.modalPrimaryAction = ACTION_TYPE.equipmentOutOfOrder;
+    this.isOpenModal = true;
+  }
+
+  generateKeepingLineModal(rma) {
+    this.rmaTrackClick('rma-keep-line');
+    this.modalRma = rma;
+    this.modalTitle = TEXT_FOR_MODAL.keepingLine;
+    this.modalContent = `${TEXT_FOR_MODAL.keepingLine}_modal_content`;
+    this.modalPrimaryAction = ACTION_TYPE.keepingLine;
```
[the remark this answers is not in the document]
Comment (ghyenne) [reply packages/manager/apps/telecom/src/app/telecom/telephony/line/assist/rma/rma.controller.js]: I fixed.

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/assist/rma/rma.controller.js [hunk 10]
```diff
@@ -1,101 +1,159 @@
...
+    this.modalPrimaryAction = ACTION_TYPE.cancelRma;
+    this.isOpenModal = true;
+  }
+
+  generateEquipmentOutOfOrderModal(rma) {
+    this.rmaTrackClick('rma-report-equipment');
+    this.modalRma = rma;
+    this.modalTitle = TEXT_FOR_MODAL.equipmentOutOfOrder;
+    this.modalContent = `${TEXT_FOR_MODAL.equipmentOutOfOrder}_modal_content`;
+    this.modalPrimaryAction = ACTION_TYPE.equipmentOutOfOrder;
```
[the remark this answers is not in the document]
Comment (ghyenne) [reply packages/manager/apps/telecom/src/app/telecom/telephony/line/assist/rma/rma.controller.js]: I fixed.

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/assist/rma/rma.controller.js (hunk 3 above)
Comment (ghyenne) [reply packages/manager/apps/telecom/src/app/telecom/telephony/line/assist/rma/rma.controller.js]: I fixed.

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/assist/rma/rma.controller.js (hunk 2 above)
Comment (ghyenne) [reply packages/manager/apps/telecom/src/app/telecom/telephony/line/assist/rma/rma.controller.js]: I fixed.

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/assist/rma/rma.constants.js (hunk 1 above)
Comment (ghyenne) [reply packages/manager/apps/telecom/src/app/telecom/telephony/line/assist/rma/rma.constants.js]: It's work with https, I fixed.

**Review by anooparveti — APPROVED** [review]

**Review by mohammed-zahaf — DISMISSED** [review]

**Review by mohammed-zahaf — APPROVED** [review]

**Review by sachinrameshn — APPROVED** [review]

**Review by anooparveti — APPROVED** [review]

**Comment by antleblanc** [comment]
Pulling translations.

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/assist/rma/rma.html [hunk 11]
```diff
@@ -117,50 +119,102 @@ <h1 data-translate="telephony_line_assist_rma_title"></h1>
...
+                            primary-action="$ctrl.executeAction($ctrl.modal.rma)"
+                            primary-label="{{ :: 'telephony_line_assist_rma_modal_yes' | translate }}"
+                            primary-disabled="$ctrl.actionDoing"
+                            secondary-action="$ctrl.resetModalInfo()"
+                            secondary-label="{{ :: 'telephony_line_assist_rma_modal_no' | translate }}"
+                            on-dismiss="$ctrl.resetModalInfo()"
+                        >
+                            <p
+                                data-translate="{{:: $ctrl.modal.content }}"
+                                data-translate-values="{ 'breakLine': '</br>' }"
```
Comment (antleblanc) [inline packages/manager/apps/telecom/src/app/telecom/telephony/line/assist/rma/rma.html:211]: **nitpick(non-blocking):** Might be interesting to split the actual translation into several chunk.

Comment (ghyenne) [reply packages/manager/apps/telecom/src/app/telecom/telephony/line/assist/rma/rma.html]: I fixed up it.

**Review by mohammed-zahaf — APPROVED** [review]

**Review by anooparveti — APPROVED** [review]

**Pull request merged**